[import-w3c-tests] Drop TestRepositories support

gsnedders opened the pull request:
#### a20c4164307a04b786e621693503957c8cdc0821
<pre>
[import-w3c-tests] Drop TestRepositories support
<a href="https://bugs.webkit.org/show_bug.cgi?id=258815">https://bugs.webkit.org/show_bug.cgi?id=258815</a>

Reviewed by Jonathan Bedard.

At this point, we&apos;re just carrying along a lot of configurability for
what is actually a single repository we support importing tests from.

Instead, hard-code the relevant data, as this makes it obvious we can
start to drop support for configuration options which are now dead
code. This also makes some of the duplicity of configurability obvious,
with &quot;paths_to_skip&quot; and &quot;paths_to_import&quot; obviously largely redundant
with import-expectations.json.

This also drops support for the &quot;import_options&quot; called
&quot;generate_git_submodules_description&quot; and &quot;generate_gitignore&quot;, and
dropping support immediately is the easiest way to get the tests to pass
with the repositories hard-coded.

* LayoutTests/imported/w3c/resources/TestRepositories: Removed.
* Tools/Scripts/webkitpy/w3c/test_downloader.py:
(TestDownloader.load_test_repositories):
(TestDownloader.update_import_expectations):
(TestDownloader._git_submodules_description): Deleted.
(TestDownloader.generate_git_submodules_description): Deleted.
(TestDownloader.generate_gitignore): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.generate_git_submodules_description_for_all_repositories):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
(TestImporterTest.test_harnesslinks_conversion):
(TestImporterTest.test_skip_test_import_download):

Canonical link: <a href="https://commits.webkit.org/270006@main">https://commits.webkit.org/270006@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25851f288c859db453391187fbb68b50ca0a5406

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23472 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1589 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24596 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21669 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23742 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3139 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24014 "Built successfully") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23715 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1134 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20311 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26230 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/23659 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/936 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21209 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27582 "Build is being retried. Recent messages:Failed to print configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; 14 flakes 118 failures; Uploaded test results; 6 flakes 111 failures; Compiled WebKit (retry)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21391 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21469 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25244 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/919 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18642 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/888 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1329 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3096 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1194 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Tools/Scripts/webkitpy/w3c/test_downloader.py:
```diff
@@ -48,10 +48,30 @@ def default_options():
+                    "conformance-checkers",
+                    "docs",
+                    "old-tests",
+                    "resources/testharness.css",
+                    "resources/testharnessreport.js",
```
This is kind of surprising to me.

gsnedders replied:
I would rather land this with no changes to the actual configuration, prior to changing anything.

gsnedders committed:
[import-w3c-tests] Drop TestRepositories support
https://bugs.webkit.org/show_bug.cgi?id=258815

Reviewed by Jonathan Bedard.

At this point, we're just carrying along a lot of configurability for
what is actually a single repository we support importing tests from.

Instead, hard-code the relevant data, as this makes it obvious we can
start to drop support for configuration options which are now dead
code. This also makes some of the duplicity of configurability obvious,
with "paths_to_skip" and "paths_to_import" obviously largely redundant
with import-expectations.json.

This also drops support for the "import_options" called
"generate_git_submodules_description" and "generate_gitignore", and
dropping support immediately is the easiest way to get the tests to pass
with the repositories hard-coded.

* LayoutTests/imported/w3c/resources/TestRepositories: Removed.
* Tools/Scripts/webkitpy/w3c/test_downloader.py:
(TestDownloader.load_test_repositories):
(TestDownloader.update_import_expectations):
(TestDownloader._git_submodules_description): Deleted.
(TestDownloader.generate_git_submodules_description): Deleted.
(TestDownloader.generate_gitignore): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.generate_git_submodules_description_for_all_repositories):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
(TestImporterTest.test_harnesslinks_conversion):
(TestImporterTest.test_skip_test_import_download):

Canonical link: https://commits.webkit.org/270006@main
